cmd/kops/upgrade_cluster.go

```diff
@@ -129,26 +130,40 @@ func (c *UpgradeClusterCmd) Run(args []string) error {
 		channelClusterSpec = &api.ClusterSpec{}
 	}
 
-	//latestKubernetesVersion, err := api.FindLatestKubernetesVersion()
-	//if err != nil {
-	//	return err
-	//}
+	var currentKubernetesVersion *semver.Version
+	{
+		sv, err := util.ParseKubernetesVersion(cluster.Spec.KubernetesVersion)
+		if err != nil {
+			glog.Warningf("error parsing KubernetesVersion %q", cluster.Spec.KubernetesVersion)
+		} else {
+			currentKubernetesVersion = sv
+		}
+	}
 
-	// So we can propose an image for the upgraded k8s version
-	upgradedKubernetesVersion := cluster.Spec.KubernetesVersion
+	proposedKubernetesVersion := api.RecommendedKubernetesVersion(channel)
 
-	if channelClusterSpec.KubernetesVersion != "" && cluster.Spec.KubernetesVersion != channelClusterSpec.KubernetesVersion {
+	// We won't propose a downgrade
+	// TODO: What if a kubernetes version is bad?
```

Comment:
Yes, we have minimum versions and required upgrades of k8s, but we need an explicit "bad list" I think